src/content/developers/docs/scaling/zk-rollups/index.md

```diff
@@ -1,34 +1,206 @@
+
+Recursive proofs, however, make it possible to finalize several blocks with one validity proof. This is because the proving circuit recursively aggregates multiple block proofs until one final proof is created. The L2 operator submits this recursive proof, and if the contract accepts it, all the relevant blocks will be finalized instantly. With recursive proofs, the number of ZK-rollup transactions that can be finalized on Ethereum at intervals increases. 
+
+### Pros and cons of zk-rollups {#zk-rollups-pros-and-cons}
+
+| Pros                                                                    | Cons                                                                                                  |
+| ------------------------------------------------------------------------| ----------------------------------------------------------------------------------------------------- |
+| Validity proofs ensure correctness of off-chain transactions and prevent operators from executing invalid state transitions. | The cost associated with computing and verifying validity proofs is substantial and can increase fees for rollup users.             |                     
+| Offers faster transaction finality as state updates are approved once validity proofs are verified on L1. | Building EVM-compatible ZK-rollups is difficult due to complexity of zero-knowledge technology.                                                     |                                                                                                         
+| Relies on trustless cryptographic mechanisms for security, not the honesty of incentivized actors as with [optimistic rollups](/developers/docs/scaling/optimistic-rollups/#optimistic-pros-and-cons)                                           | Producing validity proofs requires specialized hardware, which may encourage centralized control of the chain by a few parties.                                                            |
+| Stores data needed to recover the state on the L1 chain, which guarantees security, censorship-resistance, and decentralization. | Centralized operators (sequencers) can influence transaction ordering.                                                     |         
+| Users benefit from greater capital efficiency and can withdraw funds from L2 without delays.              | Hardware requirements may reduce the number of participants that can force the chain to make progress, increasing the risk of malicious operators freezing the rollup's state and censoring users.   | 
+| Doesn't depend on liveness assumptions and users don't have to validate the chain to protect their funds. | Some proving systems (e.g., ZK-SNARK) require a trusted setup which, if mishandled, could potentially compromise a ZK-rollup's security model.    | 
+| Better data compression can help reduce user fees.                      |                                                                                                      |
+ 
```

Comment:
I would like to see some discussion about trusted and non-trusted set ups somewhere on this page.

Reply:
e.g. https://vitalik.ca/general/2022/03/14/trustedsetup.html

Reply:
Yes, I did this in the section on ZK-STARKs.

Reply:
> e.g.
> 
> * https://vitalik.ca/general/2022/03/14/trustedsetup.html
> * https://zkproof.org/2021/06/30/setup-ceremonies/

Was writing when this comment came in, but the validity proofs section now has a part explaining trusted setups. I also included the second link in the text--looks like we'd the same thing on our minds. :)

Reply:
thanks @emmanuel-awosika those additions lgtm.